lib/Raven/Context/ServerOsContext.php

```diff
@@ -0,0 +1,207 @@
+<?php
+
+/*
+ * This file is part of Raven.
+ *
+ * (c) Sentry Team
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Raven\Context;
+
+use Symfony\Component\OptionsResolver\Exception\InvalidOptionsException;
+use Symfony\Component\OptionsResolver\Exception\UndefinedOptionsException;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * This class is a specialized implementation of the {@see Context} class that
+ * stores information about the operating system of the server.
+ *
+ * @author Stefano Arlandini <sarlandini@alice.it>
+ */
+class ServerOsContext extends Context
+{
+    /**
+     * @var OptionsResolver The options resolver
+     */
+    private $resolver;
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws UndefinedOptionsException If any of the options are not supported
+     *                                   by the context
+     * @throws InvalidOptionsException   If any of the options don't fulfill the
+     *                                   specified validation rules
+     */
+    public function __construct(array $data = [])
+    {
+        $this->resolver = new OptionsResolver();
```

Comment:
Since the option resolver seems to be used in multiple context, maybe we can refactor in the parent class?

Reply:
It's actually only used in the `RuntimeContext` and `ServerOsContext` classes. The other contexts don't need it as the data they contain is too generic to be defined as a set of options